File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
         int upperbound=first_last(nums,target,+1);
        int lowerbound=first_last(nums,target,-1);
       int[] arr=new int[2];
        arr[0]=lowerbound;
        arr[1]=upperbound;
        return arr;
        
    }
    
      private static int first_last(int[] nums, int tgt,int tog) {

        int lo=0;
        int hi=nums.length-1;
        int ans=-1;
        while (lo<=hi){
            int mid=(lo + hi)/2;
            if (nums[mid]<tgt){
                lo=mid+1;
            }else if (nums[mid]>tgt){
                hi=mid-1;
            }else {
                ans=mid;
                if (tog==-1){
                    hi=mid+tog;
                }else {
                    lo=mid+tog;
                }
            }
        }
        return ans;
    }
}

File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        vector<int> range={};
        if(nums.size()==1 && nums[0]==target)
        { range.push_back(0);
         range.push_back(0);
            return range;}
        
        
            int l=-1;
            int r=-1;
            int i=0;
            while(i<nums.size() && nums[i]<=target)
            {
                if(nums[i]==target)
                {  l=i; break;}
                i++;
            }
            if(l>-1)
            {   i=0;
                 while(i<nums.size())
            {
                if(nums[nums.size()-1-i]==target)
                {  r=nums.size()-1-i; break;}
                     i++;
            }
            }
            range.push_back(l);
         range.push_back(r);
            return range;
        }
      
};